src/metabase/models/dashboard.clj

```diff
@@ -510,19 +510,25 @@
        (set/union (serdes.util/parameters-deps parameters))))
 
 (defmethod serdes.base/serdes-descendants "Dashboard" [_model-name id]
-  (let [dashcards (db/select ['DashboardCard :card_id :parameter_mappings]
-                             :dashboard_id id)
-        dashboard (db/select-one Dashboard :id id)]
+  (let [dashcards       (db/select ['DashboardCard :card_id :parameter_mappings]
+                                   :dashboard_id id)
+        dashboard       (db/select-one Dashboard :id id)
+        parameter-cards (db/select-field :id ParameterCard :parameterized_object_type "dashboard" :parameterized_object_id id)]
     (set/union
       ;; DashboardCards are inlined into Dashboards, but we need to capture what those those DashboardCards rely on
       ;; here. So their cards, both direct and mentioned in their parameters.
-      (set (for [{:keys [card_id parameter_mappings]} dashcards
-                 ;; Capture all card_ids in the parameters, plus this dashcard's card_id if non-nil.
-                 card-id (cond-> (set (keep :card_id parameter_mappings))
-                           card_id (conj card_id))]
-             ["Card" card-id]))
+      (when dashcards
```

Comment:
Both these `when`s are unnecessary - if the `dashcards` or `parameter-cards` lists are empty, `for` returns an empty list which gets turned to an empty set.